File: src/main/java/com/morph/engine/newgui/ConsoleTextField.java
package com.morph.engine.newgui;

import com.morph.engine.input.KeyPress;
import com.morph.engine.input.StdKeyEvent;
import com.morph.engine.script.debug.Console;
import com.morph.engine.graphics.Color;
import com.morph.engine.math.Vector2f;

import static org.lwjgl.glfw.GLFW.*;

/**
 * Created on 8/2/2017.
 */
public class ConsoleTextField extends TextField {
    private Console console;

    public ConsoleTextField(String name, Console console, String text, String font, int size, Color color, Vector2f position, int depth) {
        super(name, text, font, size, color, position, depth);
        this.console = console;
    }

    public ConsoleTextField(String name, Console console, String text, String font, int size, Color color, Vector2f position) {
        super(name, text, font, size, color, position);
        this.console = console;
    }

    public ConsoleTextField(String name, Console console, String text, String font, int size, Vector2f position) {
        super(name, text, font, size, position);
        this.console = console;
    }

    public void handleGUIKeyEvent(StdKeyEvent e) {
        if (e.getAction() instanceof KeyPress && e.getKey() == GLFW_KEY_ENTER) processLine();

        super.handleGUIKeyEvent(e);
    }

    private void processLine() {
        console.readIn(text);
        clearText();
    }
}

File: src/main/kotlin/com/morph/engine/newgui/ConsoleTextField.kt
package com.morph.engine.newgui

import com.morph.engine.graphics.Color
import com.morph.engine.input.KeyPress
import com.morph.engine.input.StdKeyEvent
import com.morph.engine.math.Vector2f
import com.morph.engine.script.debug.Console
import org.lwjgl.glfw.GLFW.GLFW_KEY_ENTER

/**
 * Created on 8/2/2017.
 */
class ConsoleTextField(
        name: String,
        var console: Console,
        text: String,
        font: String,
        size: Int,
        color: Color = Color(0f, 0f, 0f),
        position: Vector2f,
        depth: Int = 0
) : TextField(name, text, font, size, color, position, depth) {
    override fun handleGUIKeyEvent(e: StdKeyEvent) {
        if (e.action === KeyPress && e.key == GLFW_KEY_ENTER) processLine()
        super.handleGUIKeyEvent(e)
    }

    private fun processLine() {
        console.readIn(text)
        clearText()
    }
}
